### PR TITLE
Refactor/timepicker

### PR DESCRIPTION
**Pull request by davodey**
Simplified the timepicker code

### CODE REVIEW

**Comment by pauljadam** [comment]
@davodey I'm not sure if aria ID references are case sensitive but the validator and axe are showing a few errors saying the id references dont exist on the page

image

I think it's because it says "itemlist" in the reference but the ID value is "itemList" with a capital L. 


The other issue is keyboard operability that matches a <select> control is missing i.e. can't re-expand the drop down after you hit escape to close it, it does not open again with the down arrow and the spacebar key like a native select control. 

Thanks!

File: timepicker/markup-kit/js/dist/main.js [hunk 1]
```diff
@@ -1,263 +1,190 @@
...
-                throwError();
-            } else {
-                clearError();
-            }
+      }
+    });
+    // closes dropdown on escape keypress
+    doc.addEventListener('keydown', function (event) {
+      if (event.key === 'Escape') {
+        dropdown.style.display = 'none';
```
Comment (jackiejakob) [inline timepicker/markup-kit/js/dist/main.js]: Need to change aria-expanded to false as well?

**Comment by jackiejakob** [comment]
@pauljadam  also, on ESC should focus move back to input?

**Comment by pauljadam** [comment]
@jackiejakob  I also noticed that keyboard trap. It should work like a native select where you can hit escape to close it or make a selection and activate the selection to close it.  Yes focus should go back on escape key.

**Comment by jackiejakob** [comment]
@pauljadam is VO reading the aria-selected for you? That was something I couldn't get to work before I handed off to Dave.

**Comment by pauljadam** [comment]
@jackiejakob I think it's a bug/lack of support in VoiceOver macOS because on iOS it's speaking the selected state to VoiceOver. One extra way to convey the state to VoiceOver macOS users could be to add some alt text for the checkmark icon e.g. role=img aria-label="Checked" and then they would at least hear checked even when they don't hear selected.

**Comment by davodey** [comment]
@pauljadam @jackiejakob I think all the errors are fixed.

**Comment by jackiejakob** [comment]
@davodey Looking good!  I saw a few minor things:

- If tab or click to blank input and then tab again or click outside of input, the inline error message will appear. Seems like it would be jarring to have it appear if I'm just tabbing through the page to see everything, but I'm open if you think we should leave it.

- A ScrollIntoView error appears in the console when I open the dropdown via down arrow from an empty input. Is this okay?

- Via keyboard, if I open the dropdown with the down arrow but then tab instead of press the down arrow to go into the dropdown, my focus goes to the dropdown list container. Then if i press down/up arrows, I'm getting some console errors. I did this by accident (I know it's standard behavior to down arrow instead of tab into the list), but thought I'd mention in case you want to fix.

**Comment by antelopeb** [comment]
@davodey -- looking good so far!

Icons are out of place on all but the first sample:
image

Also, does this support 24 hour time yet?

**Comment by davodey** [comment]
@antelopeb thank you for reminding me.  I only completed the one version.  Ill get those done.

No, it doesn't support 24 hour time, but I'll knock something out as a demo and we can program the functionality in the actual component.

**Comment by davodey** [comment]
@jackiejakob good catch.  I fixed all those issues.

**Comment by davodey** [comment]
@antelopeb I added a 24 hour time example and also fixed the read only and disabled examples.

**Pull request merged**